src/ClampUnsafeAccesses.cpp

```diff
@@ -0,0 +1,88 @@
+#include "ClampUnsafeAccesses.h"
+#include "IREquality.h"
+#include "IRMutator.h"
+#include "IRPrinter.h"
+#include "Simplify.h"
+
+namespace Halide::Internal {
+
+namespace {
+
+struct ClampUnsafeAccesses : IRMutator {
+    const std::map<std::string, Function> &env;
+    FuncValueBounds &func_bounds;
+
+    ClampUnsafeAccesses(const std::map<std::string, Function> &env, FuncValueBounds &func_bounds)
+        : env(env), func_bounds(func_bounds) {
+    }
+
+protected:
+    using IRMutator::visit;
+
+    Expr visit(const Let *let) override {
+        return visit_let<Let, Expr>(let);
+    }
+
+    Stmt visit(const LetStmt *let) override {
+        return visit_let<LetStmt, Stmt>(let);
+    }
+
+    Expr visit(const Variable *var) override {
+        if (is_inside_indexing && let_var_inside_indexing.contains(var->name)) {
+            let_var_inside_indexing.ref(var->name) = true;
+        }
+        return var;
+    }
+
+    Expr visit(const Call *call) override {
+        if (call->call_type != Call::Halide) {
+            return IRMutator::visit(call);
+        }
+
+        if (is_inside_indexing) {
+            auto bounds = func_bounds.at({call->name, call->value_index});
+            if (bounds_smaller_than_type(bounds, call->type)) {
+                // TODO(#6297): check that the clamped function's allocation bounds might be wider than its compute bounds
+
+                auto [new_args, changed] = mutate_with_changes(call->args);
+                Expr new_call = changed ? call : Call::make(call->type, call->name, new_args, call->call_type, call->func, call->value_index, call->image, call->param);
```

Comment:
(resurrecting a zombie PR...)

Uh, is this right? Shouldn't it be `!changed ? call : ...` ?

Reply:
Yeeeep, that looks like a typo. Want to try changing it to see if tests fail?